src/utils/decoders.ts

```diff
@@ -22,59 +35,68 @@ export function decodeData(
   return value;
 }
 
+const getUniqueAnonymousRecordId = getUniqueId('AnonymousRecord');
+
 function decodeRecord(
   key: string,
   recInfo: SerializedRecord,
   options: DeserializationOptions,
 ) {
   const { __record: recordName, data } = recInfo;
-  const { recordTypes = {} } = options;
+  const { recordTypes = {}, throwOnMissingRecordType = true } = options;
   const RecordType = recordTypes[recordName];
-  if (!RecordType) {
+  if (!RecordType && throwOnMissingRecordType) {
     throw new Error(`Unknown record type: ${recordName}`);
   }
 
   let decodedData: any = decodeData(key, data, options);
-  if (typeof (RecordType as any).migrate === 'function') {
+  if (!!RecordType && typeof (RecordType as any).migrate === 'function') {
     decodedData = (RecordType as any).migrate(decodedData);
   }
 
-  return new RecordType(decodedData);
+  let record;
+  if (!RecordType) {
+    // If the record type does not exist, create an AnonymousRecord
+    // that contains all of the keys in the decodedData with a default of
+    // undefined
+    const defaultsForAnonymousRecord = Object.keys(decodedData).reduce(
+      (defaults, dataKey: string) => {
+        defaults[dataKey] = undefined;
+        return defaults;
+      },
+      {} as any,
+    );
+    record = new (Record(
```

Comment:
Would be kinda cool if you cache this Record type somehow based on the keys so there is only one anon. Record type for a given "schema". Idk how much overhead it would be in code / what the perf tradeoffs would be between the two.